src/controller.js

```diff
@@ -0,0 +1,115 @@
+const bgSeaImage = document.getElementById("viewport");
+let count = 0;
+const image0 = "url('../images/water0.png')";
+const image1 = "url('../images/water1.png')";
+const messageText = document.getElementById("message-text");
+
+class Controller {
+  constructor(ship) {
+    this.ship = ship;
+    this.atSail = false;
+    this.setSail = function () {
+      if (this.atSail === true || modal.style.display === "block") {
+        return;
+      } else if (this.atSail === false) {
+        if (this.ship.remainingPort.length > 1) {
+          this.setOffSound();
+          this.renderMessage(
+            this.ship.currentPort.name,
+            this.ship.nextPort.name
+          );
+          this.atSail = true;
+          let portList = document.querySelectorAll(".port");
+          let portArray = Array.from(portList);
```

Comment:
Since this variable is only used once it doesnt really need to be declared, could just be used directly:
```
          let portArray = Array.from(document.querySelectorAll(".port"));
```